### ISSUE TITLE
Trouble understanding "continue" training

### ISSUE THREAD

**Issue by davidbelle** [facebookresearch/wav2letter#445]
Hello!

I have wav2letter running and able to transcribe wav files which is fantastic. The results aren't very accurate though, but I believe this is as a result of using Australian accents while the training data (LibriSpeech) probably is mostly american and maybe some others. I would like to "build" on top of the existing training using the "continue" flag, but when I run it I get this output 

`E1118 06:47:43.735816   203 Serial.h:74] Error while loading: failed to open file for reading
E1118 06:47:44.736126   203 Serial.h:74] Error while loading: failed to open file for reading
E1118 06:47:46.736249   203 Serial.h:74] Error while loading: failed to open file for reading
E1118 06:47:50.736377   203 Serial.h:74] Error while loading: failed to open file for reading
E1118 06:47:58.736536   203 Serial.h:74] Error while loading: failed to open file for reading
E1118 06:48:14.736699   203 Serial.h:74] Error while loading: failed to open file for reading
terminate called after throwing an instance of 'std::runtime_error'
  what():  failed to open file for reading
*** Aborted at 1574059694 (unix time) try "date -d @1574059694" if you are using GNU date ***
PC: @     0x7f92489a8428 gsignal
*** SIGABRT (@0xcb) received by PID 203 (TID 0x7f929618c600) from PID 203; stack trace: ***
    @     0x7f928e497390 (unknown)
    @     0x7f92489a8428 gsignal
    @     0x7f92489aa02a abort
    @     0x7f92492eb84d __gnu_cxx::__verbose_terminate_handler()
    @     0x7f92492e96b6 (unknown)
    @     0x7f92492e9701 std::terminate()
    @     0x7f92492e9969 __cxa_rethrow
    @           0x49c25f w2l::retryWithBackoff<>()
    @           0x419f19 main
    @     0x7f9248993830 __libc_start_main
    @           0x481549 _start
    @                0x0 (unknown)
Aborted (core dumped)`

I only have 93 files wav samples, each one ranging from 3-10 seconds long, so for now it's just to a test run to get the additional training working. There are 75 in the training list and 18 in the valid list. I have copied the LibriSpeech tutorial of how the lists should look (id location duration transcription). I have tried a few different ways of calling Test to make it work but to no avail. 

`
# Tried using the LibriSpeech dataset directory
/Train continue /root/w2l
# Tried using a flagsfile too
./Train continue --flagsfile=
`

Contents of flagsfile
`# Training config for Mini Librispeech
# Replace `[...]` with appropriate paths
--datadir=/root/w2l
--rundir=/root/w2l/save
--archdir=/root/wav2letter/tutorials/1-librispeech_clean/
--train=lists/training.lists
--valid=lists/valid.lists
--input=flac
--arch=network.arch
--tokens=/root/w2l/am/tokens.txt
--lexicon=/root/w2l/am/lexicon.txt
--criterion=ctc
--lr=0.1
--maxgradnorm=1.0
--replabel=1
--surround=|
--onorm=target
--sqnorm=true
--mfsc=true
--filterbanks=40
--nthread=4
--batchsize=4
--runname=librispeech_clean_trainlogs
--iter=25`

The audio is 8K sample rate, tried to set --samplerate=8000 but no change. I haven't found any doco that say's that's how it's done. 

Any help would be greatly appreciated :)

**Comment by tlikhomanenko** [comment]
Hi @davidbelle,

Did you find solution? When you running in the continue model the second argument should be path to the model, e.g. `./Train continue bin/model/path --flagsfile=path/to/flags`

**Comment by davidbelle** [comment]
Hi @tlikhomanenko 

Yep I got that working, thanks. 

Still seeing very inaccurate results. I've only trained 100 very short files, but I would have thought that if I decoded one of the wav files I used for training I would get an accurate transcription. I'll keep playing around, still got lots to learn. Thanks.

**Comment by samin9796** [comment]
Hi @davidbelle , I am have the same trouble like you had. Can you please share the command and the location where model file can be found?

**Comment by davidbelle** [comment]
Hi @samin9796  
It's the directory where the model files are found. 

-rw-r--r-- 1 root root 3.9K Nov 12 07:57 001_config
-rw-r--r-- 1 root root  11K Nov 12 10:24 001_log
-rw-r--r-- 1 root root  15M Nov 12 10:24 001_model_last.bin
-rw-r--r-- 1 root root  15M Nov 12 10:18 001_model_lists#dev-clean.lst.bin
-rw-r--r-- 1 root root 4.3K Nov 12 10:24 001_perf

so in my case it was ./Train continue /root/w2l/save/librispeech_clean_trainlogs --flagsfile=???

Basically when you first rain the training, you would have set a "--rundir" location in the flags file. In my case I had set this to /root/w2l/save and wav2letter generated the librispeech_clean_trainlogs directory.

**Comment by tlikhomanenko** [comment]
Hi @davidbelle,

> Still seeing very inaccurate results. I've only trained 100 very short files, but I would have thought that if I decoded one of the wav files I used for training I would get an accurate transcription. I'll keep playing around, still got lots to learn. Thanks.

Yep, the first thing to check is your loss on training data and trying to make it near zero (or at least low) and check beam-search decoder for this model on training data. This model will definitely overfit but at least you will have notion how to train it.
